components/director/internal/repo/testdb/generic_repo_update_tests.go

```diff
@@ -0,0 +1,206 @@
+package testdb
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// RepoUpdateTestSuite represents a generic test suite for repository Update and UpdateWithVersion methods of any entity that has externally managed tenants in m2m table/view.
+// This test suite is not suitable for global entities or entities with embedded tenant in them.
+type RepoUpdateTestSuite struct {
+	Name                      string
+	SQLQueryDetails           []SQLQueryDetails
+	ConverterMockProvider     func() Mock
+	RepoConstructorFunc       interface{}
+	ModelEntity               interface{}
+	DBEntity                  interface{}
+	NilModelEntity            interface{}
+	TenantID                  string
+	DisableConverterErrorTest bool
+	UpdateMethodName          string
+}
+
+// Run runs the generic repo update test suite
+func (suite *RepoUpdateTestSuite) Run(t *testing.T) bool {
+	if len(suite.UpdateMethodName) == 0 {
```

Comment:
It seems like unneeded boilerplate in the suite for most of the entities. Most of the entities has this vanilla CRUD operations which are the main focus of the generic suites. However, some entities has slightly modified Read Update or Delete operations based on any set of conditions. This results in a different SQL query and the test suites can be reused, in this case you can override the methodName.

I think that it is useful to have a reasonable default name that works well for most of the entities and override it where it is not suitable instead of redelcareing it everytime.